### PR TITLE
Remove most of mezzanine

### PR DESCRIPTION
**Pull request by Pomax**
Let me start by saying this: 

image

This is going to be a multi-step process, where we first "empty out" all the apps that are mezzanine only, or were relevant in mezzanine context only, and then once everything looks good, do an update that removes migrations for emptied out apps, so we get rid of dumb things like one the migration files in the campaign app relying on a mezzanine.page object, and despite not having its own data anymore (because they're now done through wagtail), preventing `mezzanine.pages` from getting removed.

Also note that all the things that were accessible on the old `/admin` have been removed (but not the things living on `/api`) because the `/admin` route no longer exists. There is no access to the Django admin anymore, only the wagtail CMS admin. As such, JS and template overrides we had in place for making the Django admin look/interact nicer are now wholly irr-🐘

### CODE REVIEW

**Comment by alanmoo** [comment]
>  irr-🐘

ಠ_ಠ

Anyway, given the multi-step-ness, are the warnings as part of the build log (visible on heroku when searching for the review app password) expected?

**Comment by Pomax** [comment]
ooooh yeah okay this needs fixing - we call the /api/homepage route to kick off the actual homepage data react parsing run. So..... putting that back in for now.

**Comment by Pomax** [comment]
Turned out that this was `main.js` was still set to ask about the old homepage before it did the react swapins, so I removed that check and just jump straight to react swapins, and took out the swapins for the old homepage elements, since those are generated by the CMS now.

File: network-api/networkapi/static/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+why is this directory required?
```
Comment (gvn) [inline network-api/networkapi/static/README.md]: LOL

**Comment by alanmoo** [comment]
It seems like this might be a good PR for which to migrate a fully formed database for a bit more QA before we land it. Thoughts?

File: network-api/networkapi/middleware.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-
-if settings.REFERRER_HEADER_VALUE:
-    referrer_value = settings.REFERRER_HEADER_VALUE
-
-
-class ReferrerMiddleware(MiddlewareMixin):
-    def process_response(self, request, response):
-        response['Referrer-Policy'] = referrer_value
-        return response
+from django.conf import settings
```
Comment (alanmoo) [inline network-api/networkapi/middleware.py]: Was mezzanine.conf simply importing django.conf under the hood, and then adding some stuff?

**Comment by Pomax** [comment]
So the one thing that's genuinely preventing removing mezzanine fully is the highlights model, which has a `footer` field that is of a type `mezzanine.core.fields.RichTextField`. Looking at the actual highlights we have in the system atm, none of them seem to actually be using rich text, rather they use plain text. Oddly enough, some are links, but in text form rather than genuine links, so.... maybe we can just turn this into a basic Django text field first, and then see if it needs to be rich text instead?

**Comment by Pomax** [comment]
I need to un-remove campaigns: we're using that app for petition posting, which is **fairly critical**

**Comment by alanmoo** [comment]
Highlights: we might be able to drop the footer field entirely. I think it was used when we had highlights on the homepage in a specific way, but isn't necessary anymore. Take a look at the templates that use highlights; If they don't use the footer, gut it.

Does campaigns use Mezzanine? What for?

File: network-api/networkapi/highlights/migrations/0007_nullify_homepage.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import wagtail.core.fields
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('highlights', '0006_auto_20180208_2205'),
+    ]
+
+    operations = []
```
Comment (gideonthomas) [inline network-api/networkapi/highlights/migrations/0007_nullify_homepage.py]: what is the purpose of this migration?

File: network-api/networkapi/news/migrations/0012_nullify_homepage.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('news', '0011_auto_20171012_1326'),
+    ]
+
+    operations = [
```
Comment (gideonthomas) [inline network-api/networkapi/news/migrations/0012_nullify_homepage.py]: what does this migration do?

File: network-api/networkapi/static/README.md (hunk 1 above)
Comment (gideonthomas) [reply network-api/networkapi/static/README.md]: it won't work without this directory? Probably because we set the `STATICFILES_DIR` setting. If we remove that, we won't need this folder anymore.

File: network-api/networkapi/urls.py [hunk 5]
```diff
@@ -46,12 +49,11 @@
 
     # network API routes:
 
-    url(r'^api/people/', include('networkapi.people.urls')),
+    url(r'^api/campaign/', include('networkapi.campaign.urls')),
```
Comment (gideonthomas) [inline network-api/networkapi/urls.py]: why do we still need this url set?

File: network-api/networkapi/urls.py [hunk 6]
```diff
@@ -46,12 +49,11 @@
...
+    url(r'^api/highlights/', include('networkapi.highlights.urls')),
```
Comment (gideonthomas) [inline network-api/networkapi/urls.py]: same question as above

File: network-api/networkapi/settings.py [hunk 7]
```diff
@@ -529,3 +483,6 @@ def show_toolbar(request):
 # Review apps' slack bot
 GITHUB_TOKEN = env('GITHUB_TOKEN')
 SLACK_WEBHOOK_RA = env('SLACK_WEBHOOK_RA')
+
+# REQUIRED FOR AS LONG AS MEZZANINE HAS NOT BEEN FULLY PURGED
+TESTING = False
```
Comment (gideonthomas) [inline network-api/networkapi/settings.py]: where is this used?

File: network-api/networkapi/settings.py [hunk 8]
```diff
@@ -258,18 +258,9 @@
                 'django.template.context_processors.static',
                 'django.contrib.auth.context_processors.auth',
                 'django.contrib.messages.context_processors.messages',
-                'mezzanine.conf.context_processors.settings',
-                'mezzanine.pages.context_processors.page',
-                'networkapi.context_processor.review_app',
```
Comment (gideonthomas) [inline network-api/networkapi/settings.py]: why is this removed?

File: network-api/networkapi/highlights/migrations/0007_nullify_homepage.py (hunk 3 above)
Comment (Pomax) [reply network-api/networkapi/highlights/migrations/0007_nullify_homepage.py]: it allows safe modification of 006 to remove mention of mezzanine

File: network-api/networkapi/urls.py (hunk 5 above)
Comment (Pomax) [reply network-api/networkapi/urls.py]: because the campaign app _does_ still do work for us. It's the POST handler for petition data to be forwarded to gsheets/salesforce via SQS

File: network-api/networkapi/urls.py (hunk 6 above)
Comment (Pomax) [reply network-api/networkapi/urls.py]: we use highlights in wagtail.

File: network-api/networkapi/settings.py (hunk 7 above)
Comment (Pomax) [reply network-api/networkapi/settings.py]: no idea: it's required by mezzanine for some purpose that we don't care about, since it'll get removed again when we do the second part of this work that takes out mezzanine entirely. Django/Wagtail don't need this value

File: network-api/networkapi/settings.py (hunk 8 above)
Comment (Pomax) [reply network-api/networkapi/settings.py]: it should not have been. Restoring.
